service/telemetry/telemetry.go

```diff
@@ -64,29 +67,36 @@ func New(ctx context.Context, set Settings, cfg Config) (*Telemetry, error) {
 		return nil, err
 	}
 
-	tp, err := newTracerProvider(ctx, set, cfg)
-	if err != nil {
-		return nil, err
-	}
-
-	return &Telemetry{
-		logger:         logger,
-		tracerProvider: tp,
-	}, nil
-}
-
-func newTracerProvider(ctx context.Context, set Settings, cfg Config) (*sdktrace.TracerProvider, error) {
-	opts := []sdktrace.TracerProviderOption{sdktrace.WithSampler(alwaysRecord())}
-	for _, processor := range cfg.Traces.Processors {
-		sp, err := proctelemetry.InitSpanProcessor(ctx, processor)
-		if err != nil {
-			return nil, err
-		}
-		opts = append(opts, sdktrace.WithSpanProcessor(sp))
-	}
+	sdk, err := config.NewSDK(
+		config.WithContext(ctx),
+		config.WithOpenTelemetryConfiguration(
+			config.OpenTelemetryConfiguration{
+				TracerProvider: &config.TracerProvider{
+					Processors: cfg.Traces.Processors,
+					// TODO: once https://github.com/open-telemetry/opentelemetry-configuration/issues/83 is resolved,
+					// configuration for sampler should be done here via something like the following:
+					//
+					// Sampler: &config.Sampler{
```

Comment:
After merging this PR and before resolving open-telemetry/opentelemetry-configuration/issues/83  we would have no sampling? (That's okay with me, just double checking)

Reply:
By default the parent based sampler with always sample is configured:

https://github.com/open-telemetry/opentelemetry-go/blob/1ff2e71e7d43c652f505b0680f8550ef0c863233/sdk/trace/provider.go#L484